src/conversation.rs

```diff
@@ -0,0 +1,155 @@
+use std::ffi::{CStr, CString, NulError};
+use std::ops::{Deref, DerefMut};
+
+use libc::c_void;
+use pam_sys::{set_item, PamHandle, PamItemType};
+
+use crate::{PamError, PamReturnCode};
+
+pub trait ConversationHandler<'a, C>: std::ops::Deref<Target = C> + DerefMut<Target = C> {
+    fn create(conversation: Box<C>, handle: &'a mut PamHandle) -> Self;
```

Comment:
I would call this `new` to align with existing conventions

```suggestion
    fn new(conversation: Box<C>, handle: &'a mut PamHandle) -> Self;
```